libs/gl-client/src/signer/resolve.rs

```diff
@@ -0,0 +1,103 @@
+            Message::Ping(_) => true,
+            Message::Pong(_) => true,
+            Message::SignChannelAnnouncement(_) => true,
+            Message::SignChannelUpdate(_) => true,
+            Message::SignNodeAnnouncement(_) => true,
+            // Duplicate verification with VLS, we defer to VLS
+            Message::GetChannelBasepoints(_) => true,
+            Message::ValidateCommitmentTx(_) => true,
+            Message::SignWithdrawal(_) => true,
+            Message::ReadyChannel(_) => true,
+            Message::GetPerCommitmentPoint(_) => true,
+            Message::ValidateRevocation(_) => true,
+            Message::NewChannel(_) => true,
+	    Message::SignCommitmentTx(_) => true,
+	    Message::SignGossipMessage(_) => true,
+	    Message::SignMutualCloseTx(_) => true,
+	    Message::SignMutualCloseTx2(_) => true,
+	    Message::SignRemoteCommitmentTx(_) => true,
+	    Message::SignRemoteCommitmentTx2(_) => true,
+	    // Resolution of an existing HTLC, we should never not try
+	    // to grab funds if we can.
+	    Message::SignPenaltyToUs(_) => true,
+	    Message::SignAnyPenaltyToUs(_) => true,
+	    Message::SignAnyDelayedPaymentToUs(_) => true,
+	    Message::SignAnyLocalHtlcTx(_) => true,
+	    Message::SignAnyRemoteHtlcToUs(_) => true,
+	    // Default to rejecting, punting the decision to the next
+	    // step.
+            _ => false,
+        };
+
+        // If we found a resolution, then there is no point in trying
+        // to match up further.
+        if accept {
+            log::trace!(
+                "Request {:?} resolved with no context request required",
+                req
+            );
+            return Ok(());
+        }
+
+        for cr in reqctx {
+            let accept = match (dbg!(req), dbg!(cr)) {
+                (Message::SignMessage(m1), Request::SignMessage(m2)) => {
+                    m1.message.0.clone() == m2.message.as_bytes()
+                }
+                (Message::NewChannel(m1), Request::FundChannel(m2)) => {
+                    // Different node_id? Reject!
+                    m1.node_id.0 == m2.id.as_slice()
+                    // TODO: Add `close_to` to allowlist for the close
+                    // later on
+                }
+                (Message::NewChannel(m1), Request::GlFundChannel(m2)) => {
+                    // Different node_id? Reject!
+                    m1.node_id.0 == m2.node_id.as_slice()
+                }
+                (Message::SignInvoice(_l), Request::GlCreateInvoice(_r)) => true,
+                (Message::SignInvoice(_l), Request::Invoice(_r)) => {
+                    // TODO: This could be strengthened by parsing the
+                    // invoice from `l.u5bytes` and verify the
+                    // description, amount and (maybe) payment_hash
+                    true
+                }
+                (Message::PreapproveInvoice(l), Request::Pay(r)) => {
+                    l.invstring.0 == r.bolt11.as_bytes()
```

Comment:
I notice that the amount and description are "denormalized" in the request - i.e. they exist in both the bolt11 invoice and as top-level fields request.  is it assumed that a well behaved client would always keep these identical?